### PR TITLE
 Add declaration and confirmation pages and redirects

### PR DESCRIPTION
**Pull request by russellthorn**

### CODE REVIEW

File: app/views/confirmation.scala.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+@main_template(
+    title = messages("confirmation.title"),
+    appConfig = appConfig,
+    bodyClasses = None) {
+
+	  <div class="govuk-box-highlight">
+			@components.heading("confirmation.heading", "heading-xlarge")
+      <p class="font-large">
+    		@messages("confirmation.yourRefNumber")<br>
+        <span class="bold">GBJ2123F</span>
```
Comment (ed0906) [inline app/views/confirmation.scala.html]: Does this not need to come from the back end?

File: app/views/confirmation.scala.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		<ul class="list">
+		  <li><a href="#">@messages("confirmation.download.link.PDF")</a></li>
+		  <li><a href="#">@messages("confirmation.download.link.accessibleText")</a></li>
+		</ul>
+
+		<h2 class="heading-medium">@messages("confirmation.heading2.whatNext")</h2>
+		<p>@Html(messages("confirmation.paragraph.whatNext"))</p>
+
+		<h2 class="heading-medium">@messages("confirmation.heading2.sendingSamples")</h2>
+		<p>@messages("confirmation.paragraph1.sendingSamples") GBJ2123F</p>
```
Comment (ed0906) [inline app/views/confirmation.scala.html]: Similar here, does this not need to come from the backend?

File: conf/messages.en [hunk 3]
```diff
@@ -233,3 +233,30 @@ supportingInformationDetails.checkYourAnswersLabel = Supporting information
...
+declaration.listItem2 = any supporting information and data provided on the application, or supplied after the application is submitted, can be stored on a database and published on a UK website.  This includes photographs, images, brochures and any other supporting information
+declaration.listItem3 = information provided in the confidential information section will not be published
+declaration.listItem4 = any samples you provide may be damaged during laboratory tests
+
+confirmation.title = Application complete
+confirmation.heading = Application complete
+confirmation.heading2.whatNext = What happens next
+confirmation.paragraph.whatNext = The classification team will issue your commodity code ruling within 30 to 60 days. You can find out the status of your application by checking your commodity code ruling <a href="#">account page</a>.
+confirmation.heading2.sendingSamples = Sending samples
+confirmation.paragraph1.sendingSamples = You must label your sample with the application reference number .
```
Comment (ed0906) [inline conf/messages.en]: Did you mean the space before the full stop at the end?

File: app/views/confirmation.scala.html (hunk 1 above)
Comment (ed0906) [reply app/views/confirmation.scala.html]: If its not part of the scope of this story thats fine 👍

**Comment by ed0906** [comment]
It looks like there is a missing class `DeclarationFormProvider` did this get missed off the commit by any chance?

File: test/controllers/DeclarationControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import models.NormalMode
+import pages.DeclarationPage
+import play.api.mvc.Call
+import views.html.declaration
+
+class DeclarationControllerSpec extends ControllerSpecBase {
+
+  def onwardRoute = Call("GET", "/foo")
+
+  val formProvider = new DeclarationFormProvider()
```
Comment (gokyo) [inline test/controllers/DeclarationControllerSpec.scala]: @russellthorn 
My guess is that you forgot to push the file `DeclarationFormProvider` in the package `forms` 

This is not compiling

Comment (gokyo) [reply test/controllers/DeclarationControllerSpec.scala]: I think I can fix this for @russellthorn 
But please review my changes to double check

**Comment by gokyo** [comment]
@russellthorn @pllaurado @ed0906 
In my last commit I have removed the unused form in the tests.
Could you please validate my last commit?

**Comment by russellthorn** [comment]
@pllaurado please read through epic DIT 377 before making your comments.
None of these are part of the scope
- uploadSupportingMaterialMultiple looks not fully working.
- registerBusinessRepresenting looks not fully working.
- returnSamples has no way to continue.
- previousCommodityCode has no way to continue.

File: test/controllers/DeclarationControllerSpec.scala (hunk 4 above)
Comment (gokyo) [reply test/controllers/DeclarationControllerSpec.scala]: I have done it

**Comment by pllaurado** [comment]
@russellthorn Maybe you are right, didn't read fully the epic, I just go through the actual implementation trying the functionality. It will be worth to add it into a new card.
Thanks

File: app/views/confirmation.scala.html (hunk 1 above)
Comment (ed0906) [reply app/views/confirmation.scala.html]: From the acceptance criteria "The submitted application is stored in the database of the operational service for processing"
we can just do this as part of DIT-409

Comment (russellthorn) [reply app/views/confirmation.scala.html]: "The submitted application is stored in the database of the operational service for processing" That is a different thing. That clearly states "submitted to and stored in the database" NOT to play back generated the ref number to be displayed on the confirmation.

**Pull request merged**